library/std/src/path.rs

```diff
@@ -2386,13 +2386,26 @@ impl Path {
         fs::read_dir(self)
     }
 
-    /// Returns `true` if the path points at an existing entity.
+    /// Returns `true` if the path points at an existing entity and syscall succeeds.
```

Comment:
Hm, I'm not sure if wording this in terms of syscalls is the best way to approach this, not to mention that it is not entirely clear what kind of syscall is being referenced here. This also, in some ways, leaks the implementation details which aren't necessarily true on all targets or in the future.

There's already some elaboration on additional cases where `false` can be returned in the next paragraph. I think that may be a better place for this note. For example:

> This function will traverse symbolic links to query information about the destination file. In the case where the information cannot be queried (e.g. there are broken symbolic links) this will return `false`.

Reply:
Yeah, I also felt it's not perfect but didn't come up with anything else. Now that I think of it again, maybe "and no internal failure occurs"?

>  This also, in some ways, leaks the implementation details which aren't necessarily true on all targets or in the future.

I don't think it's physically possible to check existence of a file without some kind of syscall and would be highly surprised if there was one guaranteed to never fail.

> There's already some elaboration on additional cases where `false` can be returned in the next paragraph.

I believe it'd be valuable to have at least some small head up in the summary, because the method looks simpler than it actually is. Someone could just read the description and signature, think "ah, that's obvious" and go write the code.

Reply:
I agree with @nagisa's comment and would prefer if this were not worded specifically around "if the syscall succeeds", but rather in terms of the state of the underlying system which std is providing the caller an API over.

Maybe like:

> Returns true if we can tell that the path points at an existing entity. False if it doesn't **or** if we can't tell.

Reply:
That wording sounds good.